src/boot/grub2/ostree-grub-generator

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+
+# To use a custrom script for generating grub.cfg, set the --with-grub2-mkconfig=no
+# configure switch when configuring and building OSTree.
+#
+# This script is called by ostree/src/libostree/ostree-bootloader-grub2.c whenever
+# boot loader configuration file needs to be updated. It can be used as a template
+# for a custom grub.cfg generator. What to consider when writing a custom grub.cfg
+# generator:
+#
+#   - The populate_menu() function converts boot loader entries as defined by
+#   https://www.freedesktop.org/wiki/Specifications/BootLoaderSpec/ into GRUB2
+#   menuentry sections. This is the core logic that is required by OSTree
+#   based system.
+#
+#   - Embedded systems: Be aware that this script is executed not only on a host machine by OS
+#   installer, but also on a target device, thus think about shell portability. A target device
+#   for example might be using busybox with a limited shell.
+#
+# Feel free to edit this script to fit your requirements.
+
+set -e
+
+script=$(basename ${0})
+# Atomically safe location where to generete grub.cfg when executing system upgrade.
+new_grub2_cfg=${2}
+entries_path=$(dirname $new_grub2_cfg)/entries
+
+read_config()
+{
+    config_file=${entries_path}/${1}
+    title=""
+    initrd=""
+    options=""
+    linux=""
+
+    while read -r line
+    do
+        record=$(echo ${line} | cut -f 1 -d ' ')
+        value=$(echo ${line} | cut -s -f2- -d ' ')
```

Comment:
>  At some point though I do want to revisit #228 (comment) - I didn't quite understand you reply there. The GRUB2 builtin BLS loader wouldn't have any on-target dependencies either, but it would need some changes to be possibly more flexible/configurable?

What I meant here is that this logic could be implemented entirely in ostree. Instead of working on making a GRUB2 builtin BLS loader more configurable, there could be a c++ binary (configurable) in ostree project that does what normally would be the job of "GRUB2 builtin BLS loader".

Then we could reuse the same code path what is currently used by ostree-grub-generator. Where ostree instead of executing ostree-grub-generator shell script would execute the above mentioned c++ binary.

Hopefully this clarifies it better.